lib/dialects/snowflake/query-generator.js

```diff
@@ -0,0 +1,633 @@
+      }
+
+      if (definition.includes('REFERENCES')) {
+        definition = definition.replace(/.+?(?=REFERENCES)/, '');
+        attrSql += query(`ADD FOREIGN KEY (${this.quoteIdentifier(attributeName)}) ${definition}`).replace('ALTER COLUMN', '');
+      } else {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} TYPE ${definition}`);
+      }
+
+      sql.push(attrSql);
+    }
+
+    return sql.join('');
+  }
+
+  renameColumnQuery(tableName, attrBefore, attributes) {
+    const attrString = [];
+
+    for (const attrName in attributes) {
+      const definition = attributes[attrName];
+      attrString.push(`'${attrBefore}' '${attrName}' ${definition}`);
+    }
+
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'RENAME COLUMN',
+      attrString.join(' to '),
+      ';'
+    ]);
+  }
+
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = this.parseConditionObject(smth.conditions).map(condition =>
+          `${this.jsonPathExtractionQuery(condition.path[0], _.tail(condition.path))} = '${condition.value}'`
+        );
+
+        return conditions.join(' AND ');
+      }
+      if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json functions
+        if (this._checkValidJsonStatement(smth.path)) {
+          str = smth.path;
+        } else {
+          // Also support json property accessors
+          const paths = _.toPath(smth.path);
+          const column = paths.shift();
+          str = this.jsonPathExtractionQuery(column, paths);
+        }
+
+        if (smth.value) {
+          str += util.format(' = %s', this.escape(smth.value));
+        }
+
+        return str;
+      }
+    } else if (smth instanceof Utils.Cast) {
+      if (/timestamp/i.test(smth.type)) {
+        smth.type = 'datetime';
+      } else if (smth.json && /boolean/i.test(smth.type)) {
+        // true or false cannot be casted as booleans within a JSON structure
+        smth.type = 'char';
+      } else if (/double precision/i.test(smth.type) || /boolean/i.test(smth.type) || /integer/i.test(smth.type)) {
+        smth.type = 'decimal';
+      } else if (/text/i.test(smth.type)) {
+        smth.type = 'char';
+      }
+    }
+
+    return super.handleSequelizeMethod(smth, tableName, factory, options, prepend);
+  }
+
+  /*
+  _toJSONValue(value) {
+    // true/false are stored as strings in mysql
+    if (typeof value === 'boolean') {
+      return value.toString();
+    }
+    // null is stored as a string in mysql
+    if (value === null) {
+      return 'null';
+    }
+    return value;
+  }
+  */
+
+  truncateTableQuery(tableName) {
+    return `TRUNCATE ${this.quoteTable(tableName)}`;
+  }
+
+  deleteQuery(tableName, where, options = {}, model) {
```

Comment:
nit: change this to use `Utils.joinSQLFragments`.

Reply:
yes